### PR TITLE
prov/cxi: synchronous fi_close on collective multicast

### PR DESCRIPTION
**Pull request by bulbul-hpe**
The fi_close() operation manages its internal state and return FI_SUCCESS, or a fatal error code on error.

### CODE REVIEW

File: prov/cxi/src/cxip_coll.c [hunk 1]
```diff
@@ -3073,6 +3089,7 @@ static void _curl_delete_mc_obj(struct cxip_coll_mc *mc_obj)
 quit:
 	free(url);
 	if (ret < 0) {
+		mc_obj->close_state = ret;
```
Comment (swelch) [inline prov/cxi/src/cxip_coll.c:3092]: Here and below where you set mc_obj->close_state = ret and the call free(mc_obj). Why is this being done?

Comment (bulbul-hpe) [reply prov/cxi/src/cxip_coll.c:3092]: Accessing memory after free. Works by chance but is undefined behavior. Fixing it.

**Review by heemstra — APPROVED** [review]

**Review by heemstra — CHANGES_REQUESTED** [review]
Removing approval to prevent accidental merge.

**Review by swelch — APPROVED** [review]
Thanks for the updates.

**Review by heemstra — APPROVED** [review]

**Pull request merged**